File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/GroupRenameEvent.java
package work.socialhub.kslack.entity.event;

/**
 * When a private channel is renamed, the group_rename event is sent to all connections for members of a private channel.
 * Clients can use this to update their local list of private channels.
 * <p>
 * https://api.slack.com/events/group_rename
 */
public class GroupRenameEvent implements Event {

    public static final String TYPE_NAME = "group_rename";

    private final String type = TYPE_NAME;
    private Channel channel;

    public String getType() {
        return this.type;
    }

    public Channel getChannel() {
        return this.channel;
    }

    public void setChannel(Channel channel) {
        this.channel = channel;
    }

    public static class Channel {
        private String id;
        private String name;
        private Integer created;

        public String getId() {
            return this.id;
        }

        public String getName() {
            return this.name;
        }

        public Integer getCreated() {
            return this.created;
        }

        public void setId(String id) {
            this.id = id;
        }

        public void setName(String name) {
            this.name = name;
        }

        public void setCreated(Integer created) {
            this.created = created;
        }
    }

}

File: core/src/commonMain/kotlin/work/socialhub/kslack/entity/event/GroupRenameEvent.kt
package work.socialhub.kslack.entity.event

/**
 * When a private channel is renamed, the group_rename event is sent to all connections for members of a private channel.
 * Clients can use this to update their local list of private channels.
 *
 *
 * https://api.slack.com/events/group_rename
 */
class GroupRenameEvent : Event {
    override val type: String = TYPE_NAME
    var channel: Channel? = null

    class Channel {
        var id: String? = null
        var name: String? = null
        var created: Int? = null
    }

    companion object {
        const val TYPE_NAME: String = "group_rename"
    }
}